### PR TITLE
add applyKvTransform, fix #1020

### LINKED ISSUE #1020: Handle KvCoder in SCollection#applyTransform
Right know coder inference doesn't work if the transform being applied outputs a `KV<K, V>`. We should handle this case.

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

**Comment by rculbertson** [comment]
👍

File: scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala [hunk 1]
```diff
@@ -130,8 +130,23 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    * [[SCollection]].
    */
   def applyTransform[U: ClassTag](transform: PTransform[_ >: PCollection[T], PCollection[U]])
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: I wonder if we should make this a `TypeTag` and get `K` and `V` types directly. But then we might as well fix replace `ClassTag` with `TypeTag` everywhere?

**Comment by nevillelyh** [comment]
We should fix ScioContext#customInput as well.

File: scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala (hunk 1 above)
Comment (nevillelyh) [reply scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: Gave it a try but am going to abandon this idea.
`applyTransform` is used in a lot of places and it'd require type signature change everywhere.

**Pull request merged**